### PR TITLE
support taobao mini game

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: pal/minigame/taobao_minigame.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
+/*
+ Copyright (c) 2022-2023 Xiamen Yaji Software Co., Ltd.
```
Comment (minggo) [inline pal/minigame/taobao_minigame.ts:2]: I think should use `-` to connect words for file name as others.

File: platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/Canvas.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      get: function get() {
+        return innerHeight
+      }
+    })
+  
+    return canvas
+  },
+});
+
+// 暴露全局的 canvas
```
Comment (minggo) [inline platforms/minigame/platforms/taobao-mini-game/wrapper/builtin/Canvas.js]: Please don't use chinese.

File: platforms/minigame/platforms/taobao-mini-game/wrapper/fs-utils.js [hunk 3]
```diff
@@ -0,0 +1,266 @@
+/****************************************************************************
+ Copyright (c) 2022 Xiamen Yaji Software Co., Ltd.
```
Comment (minggo) [inline platforms/minigame/platforms/taobao-mini-game/wrapper/fs-utils.js]: error date

File: platforms/minigame/common/engine/AssetManager.js [hunk 4]
```diff
@@ -21,32 +21,33 @@ if (sys.platform === sys.Platform.BAIDU_MINI_GAME) {
...
     if (REGEX.test(url)) {
         onComplete && onComplete(new Error('Can not load remote scripts'));
-    }
-    else {
+    } else {
         //TODO: Can't load scripts dynamically on Taobao platform
         if (sys.platform !== sys.Platform.TAOBAO_CREATIVE_APP) {
-            require('../../../' + url);
+            if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
+                __taobaoRequire(`../../../${url}`);
```
Comment (minggo) [inline platforms/minigame/common/engine/AssetManager.js]: Is `__taobaoRequire ` an interface provide by taobao minigame?

Comment (bofeng-song) [reply platforms/minigame/common/engine/AssetManager.js]: require has two versions of implementation, one for Taobao and one for the in-engine

File: pal/minigame/taobao_minigame.ts (hunk 1 above)
Comment (bofeng-song) [reply pal/minigame/taobao_minigame.ts:2]: The file name depends on the setting of cc.config.json, which uses the lower case of the platform name as the file name.

File: cc.config.json [hunk 5]
```diff
@@ -397,6 +397,13 @@
             "value": false,
             "internal": false
         },
+        "TAOBAO_MINIGAME": {
+            "comment": "Running in the taobao mini game.",
+            "type": "boolean",
+            "value": false,
+            "ccGlobal": true,
```
Comment (PPpro) [inline cc.config.json]: close ccGlobal, it's an option for legacy 2.x usage

Comment (PPpro) [inline cc.config.json]:
```suggestion
```

File: platforms/minigame/common/engine/AssetManager.js [hunk 6]
```diff
@@ -208,107 +207,105 @@ const downloadCCONB = (url, options, onComplete) => {
...
             }
-        }
         //TODO: Can't load scripts dynamically on Taobao platform
         if (sys.platform !== sys.Platform.TAOBAO_CREATIVE_APP) {
-            require('./' + js);
+            if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
+                __taobaoRequire(js);
+            } else {
+                require(`./${js}`);
+            }
```
Comment (PPpro) [inline platforms/minigame/common/engine/AssetManager.js]: can this be 
```js
if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
                __taobaoRequire(js);
} else if (sys.platform !== sys.Platform.TAOBAO_CREATIVE_APP) {
        
                require(`./${js}`);

}
```

File: templates/taobao-mini-game/game.ejs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                _w = canvas.height;
+                _h = canvas.width;
+            }
+        }
+        canvas.width = _w;
+        canvas.height = _h;
+    }
+
+    // Adjust initial canvas size
+    if (canvas && window.devicePixelRatio >= 2) {canvas.width *= 2; canvas.height *= 2;}
```
Comment (PPpro) [inline templates/taobao-mini-game/game.ejs:36]: do we need these for taobao iOS ?
can we remove these ?

Comment (bofeng-song) [reply templates/taobao-mini-game/game.ejs:36]: Need to test to verify

Comment (bofeng-song) [reply templates/taobao-mini-game/game.ejs:36]: We currently have no testable apps

Comment (PPpro) [reply templates/taobao-mini-game/game.ejs:36]: then remember to update this part when you get testable app

**Review by PPpro — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**